cmd/guacone/cmd/analyze.go

```diff
@@ -0,0 +1,476 @@
+  Union Analysis: Combine two SBOMs to create a unified SBOM, merging software component lists while maintaining version integrity.`,
+	Example: `
+  Ingest the SBOMs to analyze:
+  $ guacone collect files guac-data-main/docs/spdx/syft-spdx-k8s.gcr.io-kube-apiserver.v1.24.4.json
+  $ guacone collect files guac-data-main/docs/spdx/spdx_vuln.json 
+
+  Difference
+  $ guacone analyze diff --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  
+  Union
+  $ guacone analyze union --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  
+  Intersection
+  $ guacone analyze intersect --uri --sboms=https://anchore.com/syft/image/ghcr.io/guacsec/vul-image-latest-6fd9de7b-9bec-4ae7-99d9-4b5e5ef6b869,https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.4-b15339bc-a146-476e-a789-6a65e4e22e54
+  `,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		if len(args) < 1 || len(args) > 1 {
+			fmt.Println("Required 1 positional arguments, got", len(args))
+			os.Exit(1)
+		}
+
+		if args[0] != "intersect" && args[0] != "union" && args[0] != "diff" {
+			fmt.Println("Invalid positional argument. Must be one of: intersect, union or diff.")
+			os.Exit(1)
+		}
+
+		ctx := logging.WithLogger(context.Background())
+		logger := logging.FromContext(ctx)
+		httpClient := http.Client{}
+		gqlclient := graphql.NewClient(viper.GetString("gql-addr"), &httpClient)
+
+		slsas, errSlsa := cmd.Flags().GetStringSlice("slsa")
+		sboms, errSbom := cmd.Flags().GetStringSlice("sboms")
+		uri, _ := cmd.Flags().GetBool("uri")
+		purl, _ := cmd.Flags().GetBool("purl")
+
+		metadata, _ := cmd.Flags().GetBool("metadata")
+		inclSoft, _ := cmd.Flags().GetBool("incl-soft")
+		inclDeps, _ := cmd.Flags().GetBool("incl-deps")
+		inclOccur, _ := cmd.Flags().GetBool("incl-occur")
+		namespaces, _ := cmd.Flags().GetBool("namespaces")
+		id, _ := cmd.Flags().GetBool("id")
+		test, _ := cmd.Flags().GetString("test")
+
+		var graphs []graph.Graph[string, *analyzer.Node]
+		var err error
+
+		if test == "" {
+			if err = verifyAnalyzeFlags(slsas, sboms, errSlsa, errSbom, uri, purl, id); err != nil {
+				fmt.Fprintf(os.Stderr, "Error: %s", err)
+				_ = cmd.Help()
+				os.Exit(1)
+			}
+
+			//create graphs
+			graphs, err = hasSBOMToGraph(ctx, gqlclient, sboms, AnalyzeOpts{
+				Metadata: metadata, InclSoft: inclSoft, InclDeps: inclDeps, InclOccur: inclOccur, Namespaces: namespaces, URI: uri, PURL: purl, ID: id})
+
+			if err != nil {
+				logger.Fatalf("Unable to generate graphs: %v", err)
+			}
+
+		} else {
+			graphs, err = readTwoSBOM(test)
+			if err != nil {
+				logger.Fatalf("Unable to generate graphs: %v", err)
+			}
+		}
+
+		if args[0] == "diff" {
```

Comment:
would it make sense for these to be subcommands? otherwise this should be a flag maybe?